src/server/serve.rs

```diff
@@ -144,16 +150,42 @@ async fn stop_and_start_threads<'a, Ip>(
         .filter(|k| !thread_handles.contains_key(*k))
         .cloned()
         .collect();
+
+    let mut errors = AardvarkErrorList::new();
+
     for (network_name, ip) in to_start {
         let (shutdown_tx, shutdown_rx) = flume::bounded(0);
         let network_name_ = network_name.clone();
         let ns = nameservers.to_owned();
+        let addr = SocketAddr::new(ip.into(), port);
+        let udp_sock = match UdpSocket::bind(addr).await {
+            Ok(s) => s,
+            Err(err) => {
+                errors.push(AardvarkError::wrap(
+                    format!("failed to bind udp listener on {addr}"),
+                    err.into(),
+                ));
+                continue;
+            }
+        };
+
+        let tcp_sock = match TcpListener::bind(addr).await {
+            Ok(s) => s,
+            Err(err) => {
+                errors.push(AardvarkError::wrap(
+                    format!("failed to bind tcp listener on {addr}"),
+                    err.into(),
+                ));
+                continue;
+            }
+        };
```

Comment:
@mheon here I am not sure what is best here

If we fail to bind udp or tcp we will skip that network and bind neither. I am not sure if it is worth to ignore tcp failures and still listen on udp or the other way around. I feel like this is not worth the extra complexity
WDYT?

Reply:
I think skipping the network is reasonable. Honestly, I would kind of prefer to shut down Aardvark altogether if we can't take the appropriate ports?

Reply:
Well if it is the first launch then we will exit with the right errors but if we just got reloaded via SIGHUP then we just log the error, at this point it is possible that other networks were able to bind.
Exiting means we bring down everything.